python/composio/tools/local/mathematical/actions/calculator.py

```diff
@@ -19,12 +21,54 @@ class CalculatorResponse(BaseModel):
 
 class Calculator(LocalAction[CalculatorRequest, CalculatorResponse]):
     """
-    Useful to perform any mathematical calculations, like sum, minus, multiplication, division, etc.
+    Performs mathematical calculations such as addition, subtraction, multiplication, and division.
     """
 
     _tags = ["calculator"]
 
+    # Define supported operators
+    operators = {
+        ast.Add: operator.add,
+        ast.Sub: operator.sub,
+        ast.Mult: operator.mul,
+        ast.Div: operator.truediv,
+        ast.Pow: operator.pow,
+        ast.USub: operator.neg,
+        ast.UAdd: operator.pos,
+    }
+
     def execute(self, request: CalculatorRequest, metadata: Dict) -> CalculatorResponse:
-        return CalculatorResponse(
-            result=str(eval(request.operation))  # pylint: disable=eval-used
-        )
+        try:
+            # Parse the expression into an AST
+            node = ast.parse(request.operation, mode="eval").body
+            result = self._safe_eval(node)
+            return CalculatorResponse(result=str(result))
+        except Exception as e:
+            # Handle exceptions and return error messages
+            return CalculatorResponse(result=f"Error: {str(e)}")
+
+    def _safe_eval(self, node):
+        if isinstance(node, ast.Constant):  # <number>
```

Comment:
The `_safe_eval` method is handling multiple concerns. Consider refactoring it into smaller methods, each handling a specific type of AST node. This could improve readability and make the code easier to test. For example, you could have separate methods for evaluating constants, binary operations, and unary operations.